plugins/modules/aci_fabric_interface_policy_group.py

```diff
@@ -0,0 +1,297 @@
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        name=dict(type="str", aliases=["policy_group"]),
+        description=dict(type="str", aliases=["descr"]),
+        name_alias=dict(type="str"),
+        type=dict(type="str", aliases=["policy_group_type"], choices=["leaf", "spine"], required=True),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    name = module.params.get("name")
+    description = module.params.get("description")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+    policy_group_type = module.params.get("type")
+
+    if policy_group_type == "leaf":
+        policy_group_class_name = "fabricLePortPGrp"
+        policy_group_class_rn = "leportgrp-{0}".format(name)
+    else:
+        policy_group_class_name = "fabricSpPortPGrp"
+        policy_group_class_rn = "spportgrp-{0}".format(name)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fabric",
+            aci_rn="fabric",
+        ),
+        subclass_1=dict(
+            aci_class="fabricFuncP",
+            aci_rn="funcprof",
+        ),
+        subclass_2=dict(
+            aci_class=policy_group_class_name,
+            aci_rn=policy_group_class_rn,
+            module_object=name,
+            target_filter={"name": name},
+        ),
+        child_classes=[
```

Comment:
Is it necessary to specify the child classes here if they are not being managed by this module?

Reply:
Yes, child_class is not necessary.